std/haxe/coro/coroutines/ChildCoroutine.hx

```diff
@@ -0,0 +1,11 @@
+package haxe.coro.coroutines;
+
+import haxe.coro.context.Context;
+
+class ChildCoroutine<T> extends BaseCoroutine<T> {
```

Comment:
These days I'm getting pretty paranoid about classes and the `extends` keyword, so let me confirm this: is it really the case that a coroutine can be either a ChildCoroutine or a ScopeCoroutine? The distinction isn't very clear to me, so I'd like to make sure.

Reply:
Just going by names, my intuition would be that something can be both a child coroutine and a (scope coroutine or a supervisor coroutine). Although it's hard to tell because both ChildCoroutine and ScopeCoroutine don't have any distinct functionality yet, so it's more like a tag right now.